File: coupon/coupon-api/src/main/java/dev/be/coupon/api/coupon/presentation/dto/CouponUsageResponse.java
package dev.be.coupon.api.coupon.presentation.dto;

import dev.be.coupon.api.coupon.application.dto.CouponUsageResult;

import java.time.LocalDateTime;
import java.util.UUID;

public record CouponUsageResponse(
        UUID userId,
        UUID couponId,
        boolean used,
        LocalDateTime usedAt
) {
    public static CouponUsageResponse from(final CouponUsageResult result) {
        return new CouponUsageResponse(
                result.userId(),
                result.couponId(),
                result.used(),
                result.usedAt()
        );
    }
}

File: coupon/coupon-api/src/main/kotlin/dev/be/coupon/api/coupon/presentation/dto/CouponUsageResponse.kt
package dev.be.coupon.api.coupon.presentation.dto

import dev.be.coupon.api.coupon.application.dto.CouponUsageResult
import java.time.LocalDateTime
import java.util.*

data class CouponUsageResponse(
    val userId: UUID,
    val couponId: UUID,
    val used: Boolean,
    val usedAt: LocalDateTime
) {
    companion object {
        fun from(result: CouponUsageResult): CouponUsageResponse {
            return CouponUsageResponse(
                userId = result.userId,
                couponId = result.couponId,
                used = result.used,
                usedAt = result.usedAt
            )
        }
    }
}
